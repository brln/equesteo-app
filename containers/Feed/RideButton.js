import React from 'react'
import { Text, TouchableOpacity, View } from 'react-native'
import { connect } from 'react-redux'

import { brand } from '../../colors'

const RideButton = (props) => {
  return (
    <View style={{borderWidth: 3, borderColor: brand}}>
      <TouchableOpacity onPress={props.onPress} style={{alignItems: 'center', justifyContent: 'center', width: 100, height: 30, borderWidth: 2, borderColor: 'white'}}>
        <Text style={{color: 'white', textAlign: 'center'}}>{ props.currentRide ? 'Continue Ride' : 'Start Ride!'}</Text>
      </TouchableOpacity>
    </View>
  )
}

function mapStateToProps (state) {
  const currentRideState = state.get('currentRide')
  return {
    currentRide: currentRideState.get('currentRide'),
  }
}

export default connect(mapStateToProps)(RideButton)
