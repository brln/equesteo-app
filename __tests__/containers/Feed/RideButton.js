import configureStore from 'redux-mock-store'
import React from "react"
import {fromJS, Map} from "immutable"
import renderer from "react-test-renderer"
import { Provider } from 'react-redux'
import thunk from "redux-thunk"

import RideButton from '../../../containers/Feed/RideButton'


const mockStore = configureStore([thunk])

test('render', () => {
  const store = mockStore(fromJS({
    currentRide: {}
  }))

  const tree = renderer.create(
    <Provider store={store}>
      <RideButton />
    </Provider>)
  expect(tree.toJSON()).toMatchSnapshot()
})

