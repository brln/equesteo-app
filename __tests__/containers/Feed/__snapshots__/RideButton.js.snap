// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<View
  style={
    Object {
      "borderColor": "#5bc0eb",
      "borderWidth": 3,
    }
  }
>
  <View
    accessible={true}
    clickable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "white",
        "borderRadius": 4,
        "borderWidth": 2,
        "height": 30,
        "justifyContent": "center",
        "opacity": 1,
        "width": 100,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "textAlign": "center",
        }
      }
    >
      Start Ride!
    </Text>
  </View>
</View>
`;
